### PR TITLE
fix(compass-global-writes): make global writes tab not appear for readonly and config/local/admin collections COMPASS-8447

### PR DESCRIPTION
**Pull request by djechlin-mongodb**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-collection/src/components/collection-tab.tsx [hunk 1]
```diff
@@ -118,10 +118,10 @@ function WithErrorBoundary({
...
   const pluginTabs = useCollectionSubTabs();
   const connectionInfoRef = useConnectionInfoRef();
-  const isGlobalWritesSupported = useConnectionSupports(
-    connectionInfoRef.current.id,
-    'globalWrites'
-  );
+  const isGlobalWritesSupported =
+    useConnectionSupports(connectionInfoRef.current.id, 'globalWrites') &&
+    !props.isReadonly &&
+    !['config', 'local', 'admin'].includes(props.namespace.split('.')[0]);
```
Comment (djechlin-mongodb) [inline packages/compass-collection/src/components/collection-tab.tsx:124]: the logic is correct but I imagine there's a more maintainable location to put it. Suggestions welcome here.

Unfortunately I don't know the logic behind config/local/admin being not sharded -- this is just copying from data explorer code.

Comment (gribnoysup) [inline packages/compass-collection/src/components/collection-tab.tsx:124]: We use `mongodb-ns` package for this. You'd want to check the `specialish` prop on the parsed namespace here

Comment (djechlin-mongodb) [reply packages/compass-collection/src/components/collection-tab.tsx]: Done!

**Review by gribnoysup — APPROVED** [review]
I'd probably suggest adding either a plugin level unit test or an e2e test for this, but otherwise code looks good!

**Comment by djechlin-mongodb** [comment]
> Member

Looks like there's no tests set up on this already so there's a bit of work to scaffold them. I discussed with Paula, we think it's just a lot to set up right now so going with merging.

**Comment by gribnoysup** [comment]
> we think it's just a lot to set up right now so going with merging.

Fair enough, but FWIW the e2e one is then probably the easiest to add, you would only need to open collection by name in any existing e2e test and check that the tab is not there, not extra setup needed

**Pull request merged**